File: src/c/ch01/1.1/l1/demo_bases.c
#include <stdio.h>
#include <stdint.h>

int main(void) {
    unsigned int n = 42;

    /* Print same value in decimal, octal, hex */
    printf("Decimal: %u\n", n);
    printf("Octal  : %o\n", n);  /* base 8 */
    printf("Hex    : %X\n", n);  /* base 16 (uppercase) */

    /* Show literal forms: C supports octal (0 prefix) and hex (0x). */
    unsigned int oct_lit = 052;   /* 42 in octal */
    unsigned int hex_lit = 0x2A;  /* 42 in hex */
    printf("Octal literal 052 -> %u\n", oct_lit);
    printf("Hex literal 0x2A -> %u\n", hex_lit);

    return 0;
}

File: src/erlang/ch01/1.1/l1/src/demo_bases.erl
%% Print decimal, octal, hex representations for a given integer.
-module(demo_bases).
-export([run/0]).

run() ->
    N = 42,
    Dec = integer_to_list(N),
    Oct = integer_to_list(N, 8),
    Hex = string:uppercase(integer_to_list(N, 16)),
    io:format("Decimal: ~s~n", [Dec]),
    io:format("Octal  : ~s~n", [Oct]),
    io:format("Hex    : ~s~n", [Hex]),

    %% Show literals via parsing strings (Erlang has only decimal literals).
    OctLit  = list_to_integer("52", 8),    % 52_8  = 42_10
    HexLit  = list_to_integer("2A", 16),   % 2A_16 = 42_10
    io:format("Octal literal 52 -> ~p~n", [OctLit]),
    io:format("Hex literal 2A -> ~p~n", [HexLit]).
